IdentityCore/src/parameters/MSIDInteractiveTokenRequestParameters.m

```diff
@@ -0,0 +1,112 @@
+
+- (instancetype)initWithAuthority:(MSIDAuthority *)authority
+                      redirectUri:(NSString *)redirectUri
+                         clientId:(NSString *)clientId
+                           scopes:(NSOrderedSet<NSString *> *)scopes
+                       oidcScopes:(NSOrderedSet<NSString *> *)oidScopes
+             extraScopesToConsent:(NSOrderedSet<NSString *> *)extraScopesToConsent
+                    correlationId:(NSUUID *)correlationId
+                   telemetryApiId:(NSString *)telemetryApiId
+                    brokerOptions:(MSIDBrokerInvocationOptions *)brokerOptions
+                      requestType:(MSIDRequestType)requestType
+              intuneAppIdentifier:(NSString *)intuneApplicationIdentifier
+                            error:(NSError **)error
+{
+    self = [super initWithAuthority:authority
+                        redirectUri:redirectUri
+                           clientId:clientId
+                             scopes:scopes
+                         oidcScopes:oidScopes
+                      correlationId:correlationId
+                     telemetryApiId:telemetryApiId
+                intuneAppIdentifier:intuneApplicationIdentifier
+                        requestType:requestType
+                              error:error];
+
+    if (self)
+    {
+        _extraScopesToConsent = [extraScopesToConsent msidToString];
+        _brokerInvocationOptions = brokerOptions;
+        _enablePkce = YES;
+    }
+
+    return self;
+}
+
+- (NSOrderedSet *)allAuthorizeRequestScopes
+{
+    NSMutableOrderedSet *requestScopes = [[self.allTokenRequestScopes msidScopeSet] mutableCopy];
+    NSOrderedSet *extraScopes = [self.extraScopesToConsent msidScopeSet];
+
+    if (extraScopes)
+    {
+        [requestScopes unionOrderedSet:extraScopes];
+    }
+    return requestScopes;
+}
+
+- (NSDictionary *)allAuthorizeRequestExtraParameters
+{
+    NSMutableDictionary *authorizeParams = [[NSMutableDictionary alloc] initWithDictionary:self.appRequestMetadata];
+    
+    if (self.extraAuthorizeURLQueryParameters && self.extraAuthorizeURLQueryParameters.count > 0)
+    {
+        [authorizeParams addEntriesFromDictionary:self.extraAuthorizeURLQueryParameters];
+    }
+    
+    if (self.extraURLQueryParameters && self.extraURLQueryParameters.count > 0)
+    {
+        [authorizeParams addEntriesFromDictionary:self.extraURLQueryParameters];
+    }
+    
+    return authorizeParams;
+}
+
+- (BOOL)validateParametersWithError:(NSError **)error
+{
+    BOOL result = [super validateParametersWithError:error];
+
+    if (!result)
+    {
+        return NO;
+    }
+
+    
+    if (self.claimsRequest.hasClaims && self.allAuthorizeRequestExtraParameters[MSID_OAUTH2_CLAIMS])
+    {
+        MSIDFillAndLogError(error, MSIDErrorInvalidDeveloperParameter, @"Duplicate claims parameter is found in extraQueryParameters. Please remove it.", nil);
+        return NO;
+    }
+
+    return YES;
+}
```

Comment:
Can we just set return variable and return that instead.

Reply:
Addressed.